packages/mendel-pipeline/src/cache/server.js

```diff
@@ -63,30 +63,23 @@ class CacheServer extends EventEmitter {
     }
 
     initCache() {
-        this.cache.on('doneEntry', entry => this.sendEntry(entry));
-        this.cache.on('entryRemoved', id => this.signalRemoval(id));
+        this.cacheManager.on('doneEntry', (cache, entry) => {
+            this.clients
+                .filter(client => client.environment === cache.environment)
+                .forEach(client => this._sendEntry(client, cache.size(), entry));
+        });
+        this.cacheManager.on('entryRemoved', id => this.signalRemoval(id));
```

Comment:
Should the signal removal also be per environment? I know most cases are because of the file being removed from the FS in the source folder. But it could also be a dependency removed by GST that only happens on environments the GST is on...  just in case, I guess.